### PR TITLE
fix(alerts&reports): add celery soft timeout support

### PR DESCRIPTION
**Pull request by dpgaspar**
### SUMMARY
Handles specifically celery's `SoftTimeoutExceeded` exception on alerts and reports. 

On alerts:
- Adds a specific message for this event on alerts, this was handled, but caught on a DBAPI broad exception catch.

On reports:
- screenshots swallow all webdriver exceptions and return `None`, added an extra broad exception that can handle `SoftTimeoutExceeded`, other exceptions can occur and these were not handled.

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/reports/commands/exceptions.py [hunk 1]
```diff
@@ -155,6 +155,10 @@ class AlertQueryError(CommandException):
     message = _("Alert found an error while executing a query.")
 
 
+class AlertQueryTimeout(CommandException):
+    message = _("A timeout occurred while executing the query.")
```
Comment (willbarrett) [inline superset/reports/commands/exceptions.py:159]: As this can happen during either the query or the thumbnail computation, perhaps we should reword this as "A timeout occurred while executing the background job." - is this the same message that would trigger when a Report times out?

Comment (dpgaspar) [reply superset/reports/commands/exceptions.py:159]: not the same message, this is really specific for queries

Comment (dpgaspar) [reply superset/reports/commands/exceptions.py:159]: created a specific exception for screenshots timeout also

File: superset/reports/commands/alert.py [hunk 2]
```diff
@@ -112,9 +128,12 @@ def _is_validator_operator(self) -> bool:
...
         )
 
-    def validate(self) -> None:
+    def _execute_query(self) -> pd.DataFrame:
         """
-        Validate the query result as a Pandas DataFrame
+        Executes the actual alert SQL query template
+
+        :return: A pandas dataframe
+        :raises: AlertQueryTimeout, AlertQueryError
```
Comment (villebro) [inline superset/reports/commands/alert.py:136]: same here

File: superset/reports/commands/execute.py [hunk 3]
```diff
@@ -172,9 +174,14 @@ def _get_screenshot(self) -> ScreenshotData:
...
-            user=user, cache=thumbnail_cache, force=True,
-        )
+        try:
+            image_data = screenshot.get_screenshot(user=user)
+        except SoftTimeLimitExceeded as ex:
+            raise ReportScheduleScreenshotTimeout()
+        except Exception as ex:
+            raise ReportScheduleScreenshotFailedError(
+                f"Failed taking a screenshot {str(ex)}"
+            )
```
Comment (villebro) [inline superset/reports/commands/execute.py:184]: I'm stealing codecov's comment here, but could we add a test for this broad exception case, too?

**Review by villebro — APPROVED** [review]
LGTM - one minor docstring nit + would be nice to reach full coverage on the exception tests.

**Review by villebro — APPROVED** [review]
